File: app/src/main/java/com/jpmorgan/typicode/activities/AlbumActivity.java
package com.jpmorgan.typicode.activities;

import android.support.v7.app.AppCompatActivity;

public abstract class AlbumActivity extends AppCompatActivity {

    private static final int STATUS_STARTED = 1;
    private static final int STATUS_RESUMED = 2;
    private static final int STATUS_PAUSED = 3;
    private static final int STATUS_STOPPED = 4;
    protected int currentStatus;

    protected void onStart() {
        super.onStart();
        currentStatus = STATUS_STARTED;
    }

    protected void onPause() {
        super.onPause();
        currentStatus = STATUS_PAUSED;
    }

    protected void onResume() {
        super.onResume();
        currentStatus = STATUS_RESUMED;
    }

    protected void onStop() {
        super.onStop();
        currentStatus = STATUS_STOPPED;
    }


}

File: app/src/main/java/com/jpmorgan/typicode/activities/AlbumActivity.kt
package com.jpmorgan.typicode.activities

import android.support.v7.app.AppCompatActivity

abstract class AlbumActivity : AppCompatActivity() {
    var currentStatus: Int = 0

    override fun onStart() {
        super.onStart()
        currentStatus = STATUS_STARTED
    }

    override fun onPause() {
        super.onPause()
        currentStatus = STATUS_PAUSED
    }

    override fun onResume() {
        super.onResume()
        currentStatus = STATUS_RESUMED
    }

    override fun onStop() {
        super.onStop()
        currentStatus = STATUS_STOPPED
    }

    companion object {

        const val STATUS_STARTED = 1
        const val STATUS_RESUMED = 2
        const val STATUS_PAUSED = 3
        const val STATUS_STOPPED = 4
    }


}
